# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/mesa/hotfix-wl-surface-hack.patch.ia64
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- mesa-25.3.1/src/egl/drivers/dri2/platform_wayland.c.vanilla	2025-12-15 20:06:37.564060297 +0100
+++ mesa-25.3.1/src/egl/drivers/dri2/platform_wayland.c	2025-12-15 20:07:19.964502357 +0100
@@ -465,19 +465,8 @@
 get_wayland_surface(struct dri2_egl_surface *dri2_surf,
                     struct wl_egl_window *window)
 {
-   struct wl_surface *base_surface;
-
-   /* Version 3 of wl_egl_window introduced a version field at the same
-    * location where a pointer to wl_surface was stored. Thus, if
-    * window->version is dereferenceable, we've been given an older version of
-    * wl_egl_window, and window->version points to wl_surface */
-   if (_eglPointerIsDereferenceable((void *)(window->version)))
-      base_surface = (struct wl_surface *)window->version;
-   else
-      base_surface = window->surface;
-
    return loader_wayland_wrap_surface(&dri2_surf->wayland_surface,
-                                      base_surface,
+                                      window->surface,
                                       dri2_surf->wl_queue);
 }
 
